sinks/xray/xray.go

```diff
@@ -80,7 +80,7 @@ type XRaySpanSink struct {
 var _ sinks.SpanSink = &XRaySpanSink{}
 
 // NewXRaySpanSink creates a new instance of a XRaySpanSink.
-func NewXRaySpanSink(daemonAddr string, sampleRatePercentage int, commonTags map[string]string, annotationTags []string, log *logrus.Logger) (*XRaySpanSink, error) {
+func NewXRaySpanSink(daemonAddr string, sampleRatePercentage float64, commonTags map[string]string, annotationTags []string, log *logrus.Logger) (*XRaySpanSink, error) {
```

Comment:
Does this irritate the `math.MaxUint32` below? (otherwise, it looks pretty simple)

Reply:
It actually doesn't! (it compiles, and I even test-logged the sample value).

If you're curious why it works: despite the semantic significance, math.MaxUint32 is actually defined as an untyped constant (`MaxUint8  = 1<<8 - 1`), so to the compiler, it's the same as multiplying a float64 by a hard-coded, untyped numeric literal.

Reply:
That was my suspicion, excellent!